ionoscloud/data_source_dbaas_redisdb_replicaset.go

```diff
@@ -0,0 +1,208 @@
+package ionoscloud
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	redisdb "github.com/ionos-cloud/sdk-go-dbaas-redis"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils/constant"
+)
+
+func dataSourceDBaaSRedisDBReplicaSet() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceDBaaSRedisDBReplicaSetRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Description: "The ID of the Redis Replicaset.",
+				Optional:    true,
+			},
+			"location": {
+				Type:             schema.TypeString,
+				Description:      "The cluster location",
+				Optional:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice(constant.MariaDBClusterLocations, false)),
+			},
+			"display_name": {
+				Type:        schema.TypeString,
+				Description: "The display name of the Redis Replicaset.",
+				Optional:    true,
+			},
+			"dns_name": {
+				Type:        schema.TypeString,
+				Description: "The DNS name of the Redis Replicaset.",
+				Computed:    true,
+			},
+			"connections": {
+				Type:        schema.TypeList,
+				Description: "The network connection for your Replicaset. Only one connection is allowed.",
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cidr": {
+							Type:        schema.TypeString,
+							Description: "The IP and subnet for your Replicaset.",
+							Computed:    true,
+						},
+						"datacenter_id": {
+							Type:        schema.TypeString,
+							Description: "The datacenter to connect your Replicaset to.",
+							Computed:    true,
+						},
+						"lan_id": {
+							Type:        schema.TypeString,
+							Description: "The numeric LAN ID to connect your Replicaset to.",
+							Computed:    true,
+						},
+					},
+				},
+			},
+			"credentials": {
+				Type:        schema.TypeList,
+				Description: "The credentials for your Replicaset.",
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"password": {
```

Comment:
@avirtopeanu-ionos not sure if this should be included here since we don't read anything about the password, add this `TODO` please: `TODO -- Check if 'password' needs to be here or not`

Reply:
you're correct, no need to add TODO, I'll remove this attribute